packages/x-date-pickers-pro/src/SingleInputDateRangeField/useSingleInputDateRangeField.ts

```diff
@@ -69,5 +69,5 @@ export const useSingleInputDateRangeField = <TDate, TChildProps extends {}>({
     fieldValueManager: dateRangeFieldValueManager,
     validator: validateDateRange,
     supportedDateSections: ['year', 'month', 'day'],
-  });
+  }) as any;
```

Comment:
I'm really not a fan of this solution.
Suggestions on how to improve/ditch it are very welcome.

Reply:
@alexfauquette Maybe you would have better ideas on how to avoid the [mentioned](https://github.com/mui/mui-x/pull/6322#issuecomment-1278853787) `TS` errors?

Reply:
I proposed a fix on https://github.com/mui/mui-x/pull/6322/files#r995675216

Small tip, if you explicitly pass the generics, you get a more precise error.
For instance for the `useDateField` hook

```tsx
  return useField<
      TDate | null,
      TDate,
      FieldSection,
      TChildProps,
      UseDateFieldDefaultizedProps<TDate>
      >({
    inputRef,
```

Reply:
Thank you for a great tip 👍 
Way easier than reverse engineering cryptic TS errors. 😆